`onBeforeRender` in `BatchedMesh` causes TS error

- `three` version: 0.139.2
- `@types/three` version: 0.139.0
- `three-stdlib` version: 2.28.0

### Problem description:

In a project that has `three-stdlib` as a dependency I see these errors when running TypeScript:

```
node_modules/.pnpm/three-stdlib@2.28.0_three@0.139.2/node_modules/three-stdlib/objects/BatchedMesh.d.ts:37:5 - error TS2425: Class 'Mesh<BufferGeometry, Material>' defines instance member property 'onBeforeRender', but extended class 'BatchedMesh' defines it as instance member function.

37     onBeforeRender(): void;
       ~~~~~~~~~~~~~~

node_modules/.pnpm/three-stdlib@2.28.0_three@0.139.2/node_modules/three-stdlib/objects/BatchedMesh.d.ts:38:5 - error TS2425: Class 'Mesh<BufferGeometry, Material>' defines instance member property 'onAfterRender', but extended class 'BatchedMesh' defines it as instance member function.

38     onAfterRender(): void;
       ~~~~~~~~~~~~~
```

### Relevant code:

I'm just importing the library, not using it yet. That is enough to trigger the `tsc` error.

### Suggested solution:

I noticed you have `@ts-ignore` above [`onBeforeRender` in `BatchedMesh`](https://github.com/pmndrs/three-stdlib/blob/341e9025f356f7d0186453c6a12876ec74967049/src/objects/BatchedMesh.ts#L384). In the generated `BatchedMesh.d.ts` these `@ts-ignore` comments get stripped:

```
import { Matrix4, Mesh, BufferGeometry, Material, DataTexture, IUniform } from 'three';
declare class BatchedMesh extends Mesh<BufferGeometry, Material> {
    // ... (other methods omitted)
    onBeforeRender(): void;
    onAfterRender(): void;
}
export { BatchedMesh };
```

Since the `@ts-ignore` comment isn't helping here, we have to actually fix the original type error. To fix it we could either make `onBeforeRender` a property in `three-stdlib` or make it a method in `@types/three`. It would be nice to know why `@types/three` defined [`onBeforeRender` as a property](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/three/src/core/Object3D.d.ts#L238) in the first place though.

--- a/src/objects/BatchedMesh.ts
+++ b/src/objects/BatchedMesh.ts
@@ -123,6 +123,20 @@ class BatchedMesh extends Mesh<BufferGeometry, Material> {
 
     this._initMatricesTexture()
     this._initShader()
+
+    this.onBeforeRender = function () {
+      if (this.material.defines) {
+        this.material.defines.BATCHING = true
+      }
+
+      // @TODO: Implement frustum culling for each geometry
+    }
+
+    this.onAfterRender = function () {
+      if (this.material.defines) {
+        this.material.defines.BATCHING = false
+      }
+    }
   }
 
   _initMatricesTexture() {
@@ -379,22 +393,6 @@ class BatchedMesh extends Mesh<BufferGeometry, Material> {
     this._matricesTexture = null
 
     return this
-  }
-
-  // @ts-ignore
-  onBeforeRender() {
-    if (this.material.defines) {
-      this.material.defines.BATCHING = true
-    }
-
-    // @TODO: Implement frustum culling for each geometry
-  }
-
-  // @ts-ignore
-  onAfterRender() {
-    if (this.material.defines) {
-      this.material.defines.BATCHING = false
-    }
   }
 }
 

